plugins/gossip/plugin.go

```diff
@@ -138,11 +143,40 @@ func configureMessageLayer() {
 		}
 
 		msg := cachedMsgEvent.Message.Unwrap()
+
+		// do not gossip requested messages
+		requestedMsgMutex.Lock()
+		if _, exist := requestedMsgs[msg.ID()]; exist {
+			delete(requestedMsgs, msg.ID())
+			return
```

Comment:
You forgot to unlock the lock here. That is the problem of having to deal with the lock at this level. It would be less error prone if the requestedMsgs would be handling the lock internally.